homeassistant/components/snooz/fan.py

```diff
@@ -0,0 +1,124 @@
+"""Fan representation of a Snooz device."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from typing import Any
+
+from pysnooz.api import UnknownSnoozState
+from pysnooz.commands import (
+    SnoozCommandData,
+    SnoozCommandResultStatus,
+    set_volume,
+    turn_off,
+    turn_on,
+)
+
+from homeassistant.components.fan import ATTR_PERCENTAGE, FanEntity, FanEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_OFF, STATE_ON
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.restore_state import RestoreEntity
+
+from .const import DOMAIN
+from .models import SnoozConfigurationData
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Snooz device from a config entry."""
+
+    data: SnoozConfigurationData = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities([SnoozFan(hass, data)])
+
+
+class SnoozFan(FanEntity, RestoreEntity):
+    """Fan representation of a Snooz device."""
+
+    def __init__(self, hass, data: SnoozConfigurationData) -> None:
+        """Initialize a Snooz fan entity."""
+        self.hass = hass
```

Comment:
```suggestion
    def __init__(self, data: SnoozConfigurationData) -> None:
        """Initialize a Snooz fan entity."""
```

Don't pass in `hass` here as it will automatically be set when the entity is added to Home Assistant

Reply:
https://github.com/home-assistant/core/blob/b7e84543c1d5265b51dbebe4a9cc87f1e5322a6b/homeassistant/helpers/entity.py#L750